### ISSUE TITLE
Unable to build dependencies.

### ISSUE THREAD

**Issue by jp80** [flare-eng/coston#5]
This will not build against the latest version of node (15.5.1)

used "n" to select 10.22.0 and everything was fine.

Also when trying truffle migrate:

```
truffle migrate --network coston --reset

Compiling your contracts...

> Everything is up to date, there is nothing to compile.

Error: PollingBlockTracker - encountered an error while attempting to update latest block:
Error: socket hang up
    at createHangUpError (_http_client.js:332:15)
    at TLSSocket.socketOnEnd (_http_client.js:435:23)
    at TLSSocket.emit (events.js:203:15)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
    at PollingBlockTracker._performSync (/home/jp/Work/Crypto/flare/coston/node_modules/eth-block-tracker/src/polling.js:51:24)
    at process._tickCallback (internal/process/next_tick.js:68:7)


truffle console --network coston
Error: PollingBlockTracker - encountered an error while attempting to update latest block:
Error: socket hang up
    at createHangUpError (_http_client.js:332:15)
    at TLSSocket.socketOnEnd (_http_client.js:435:23)
    at TLSSocket.emit (events.js:203:15)
    at TLSSocket.EventEmitter.emit (domain.js:448:20)
    at endReadableNT (_stream_readable.js:1145:12)
    at process._tickCallback (internal/process/next_tick.js:63:19)
    at PollingBlockTracker._performSync (/home/jp/Work/Crypto/flare/coston/node_modules/eth-block-tracker/src/polling.js:51:24)
    at process._tickCallback (internal/process/next_tick.js:68:7)

```

Has the costonRPC endpoint changed / broken?

**Comment by sprwn** [comment]
We implemented stricter rate-limiting of the testnet at that time due to increased usage from the Flare Finance beta, causing the socket hang-up you experienced at that time.

**Comment by xrpdevs** [comment]
Hey thanks for getting back to me ("jp80") here.

I should have updated here to say that my issues were resolved after making changes to my build environment. Whether or not this was the fix or just the time difference between trying then and trying the next time is uncertain.

If anyone is interested, I started a project https://github.com/xrpdevs/FlareNetMessenger which an Android app using the web3j library and a smart contract. It's a basic wallet and address-to-address private messaging client running on coston.

The code is VERY hacky and I'm doing network IO on the UI thread still whilst learning the ropes but hopefully it could provide a starting point for anyone who also wants to start learning.
